typed-racket-lib/typed-racket/rep/type-rep.rkt

```diff
@@ -579,6 +579,17 @@
   [#:frees (λ (f) (make-invariant (f value)))]
   [#:key 'continuation-mark-key])
 
+;; Newtype
+;; comes from define-new-subtype
+;; nm: a symbol representing the name of the Newtype
+;; id: a symbol created with gensym
+;; ty: a type for the representation
+(def-type Newtype ([nm symbol?] [id symbol?] [ty Type/c])
```

Comment:
I've been thinking about this more, and even though it isn't related to the name `define-new-subtype`, would `Distinction` be a good name for the internal `type-rep.rkt` type? (idea from @stamourv https://github.com/racket/typed-racket/pull/149#issuecomment-110394335)